### PR TITLE
make plot_holiday work with prophet holiday format

### LINKED ISSUE #702: Change dataframe format in `plot_holidays`
### 🚀 Feature Request

Change dataframe format in `plot_holidays`

### Motivation

Make method more convenient for parameters selection

### Proposal

Make the format of `holidays` dataframe  the same as in Prophet [see](https://facebook.github.io/prophet/docs/seasonality,_holiday_effects,_and_regressors.html)

### Test cases

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [ ] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by iKintosh**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes
Make plot_holiday work with prophet holiday format.

## Closing issues
Closes #702

### CODE REVIEW

**Comment by iKintosh** [comment]
```python
import pandas as pd
from etna.datasets import TSDataset
from etna.analysis import plot_holidays

ts = TSDataset(data_in_etna_format, freq="D")
new_years = pd.DataFrame({
  'holiday': 'New Year',
  'ds': pd.to_datetime(['2015-01-01', '2016-01-01', '2017-01-01', '2018-01-01', '2019-01-01']),
  'lower_window': 10,
  'upper_window': 10,
})
holidays = pd.concat((new_years,))

plot_holidays(ts, holidays=holidays, segments=["Finland_KaggleMart_Kaggle Hat"], columns_num=1, figsize=(10,7))
```

Снимок экрана 2022-05-24 в 16 40 00

**Comment by iKintosh** [comment]
I suggest we add converter function from old etna format to current (prophet) format.

File: etna/analysis/plotters.py [hunk 1]
```diff
@@ -1312,12 +1322,61 @@ def _create_holidays_df(country_holidays: Type["holidays_lib.HolidayBase"], time
...
+    if as_is:
+        dt = holidays_df.index.intersection(holidays.index)
+        holidays_df.loc[dt, :] = holidays.loc[dt, :]
+        return holidays_df
+
+    for name in holidays["holiday"].unique():
+        ds = holidays[holidays["holiday"] == name]["ds"]
+        dt = [ds]
+        if "upper_window" in holidays.columns:
+            ds_upper_bound = ds + pd.to_timedelta(
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1338]: Will it work correctly if `ts` will have 15-minute frequency for example?

Try:
```python
pd.to_timedelta(10, unit="15T")
```

It fails.

We should add smth like these into tests.

File: etna/analysis/plotters.py [hunk 2]
```diff
@@ -1312,12 +1322,61 @@ def _create_holidays_df(country_holidays: Type["holidays_lib.HolidayBase"], time
...
+                    break
+                dt.append(ds_add)
+                i += 1
+        dt = pd.concat(dt)
+        dt = holidays_df.index.intersection(dt)
+        holidays_df.loc[dt, name] = 1
+    return holidays_df
+
+
 def plot_holidays(
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1373]: We should add Raises block for error with as_is.

File: etna/analysis/plotters.py [hunk 3]
```diff
@@ -1336,27 +1395,22 @@ def plot_holidays(
 
         * if str, then this is code of the country in `holidays <https://pypi.org/project/holidays/>`_ library;
 
-        * | if DataFrame, then dataframe with holidays is expected to have timestamp index with holiday names columns.
-          | In a holiday column values 0 represent absence of holiday in that timestamp, 1 represent the presence.
+        * if DataFrame, then dataframe is expected to be in prophet`s holiday format;
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1398]: I think we should add here some info about as_is logic.

File: etna/analysis/plotters.py (hunk 1 above)
Comment (iKintosh) [reply etna/analysis/plotters.py:1342]: I fixed it and added test for this case

File: etna/analysis/plotters.py (hunk 2 above)
Comment (iKintosh) [reply etna/analysis/plotters.py:1377]: What do you mean? 
I've done it
`raise ValueError("Parameter `as_is` should be used with `holiday`: pd.DataFrame, not string.")`

File: etna/analysis/plotters.py (hunk 3 above)
Comment (iKintosh) [reply etna/analysis/plotters.py:1402]: Added on line 1409

File: etna/analysis/plotters.py (hunk 2 above)
Comment (Mr-Geekman) [reply etna/analysis/plotters.py:1369]: I mean Raises block in documentation:
```
Raises
------
ValueError:
    error description
```

File: etna/analysis/plotters.py (hunk 3 above)
Comment (Mr-Geekman) [reply etna/analysis/plotters.py:1394]: User starts reading documentation, during the reading about `holidays` param he fixed in his head the expected format, but few lines later there is a description of `as_is` format that discovers that `holidays` can be in the other format. 

I think that documentation of `holidays` should cover that its behavior depends on `as_is` parameter and explain it. I don't think that description of format should be spread across two parameters: `holidays` and `as_is`.

File: tests/test_analysis/test_plotters.py [hunk 4]
```diff
@@ -197,3 +199,72 @@ def test_validate_intersecting_segments_ok(fold_numbers):
...
+    assert df.sum().sum() == 7
+
+
+def test_create_holidays_df_as_is(simple_df):
+    holidays = pd.DataFrame(index=pd.date_range(start="2020-01-07", end="2020-01-10"), columns=["Christmas"], data=1)
+    df = _create_holidays_df(holidays, simple_df.index, as_is=True)
+    assert df.sum().sum() == 4
+
+
+def test_create_holidays_df_non_day_freq():
```
Comment (Mr-Geekman) [inline tests/test_analysis/test_plotters.py:265]: Try use not only "H" but "15T" we can have digits in our frequency.

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: etna/analysis/plotters.py [hunk 5]
```diff
@@ -1312,12 +1322,47 @@ def _create_holidays_df(country_holidays: Type["holidays_lib.HolidayBase"], time
...
+        holidays_df.loc[dt, :] = holidays.loc[dt, :]
+        return holidays_df
+
+    holidays_df = pd.DataFrame(index=index, columns=holidays["holiday"].unique(), data=False)
+    for name in holidays["holiday"].unique():
+        freq = pd.infer_freq(index)
+        ds = holidays[holidays["holiday"] == name]["ds"]
+        dt = [ds]
+        if "upper_window" in holidays.columns:
+            periods = holidays[holidays["holiday"] == name]["upper_window"].fillna(0).tolist()[0] + 1
```
[the remark this answers is not in the document]
Comment (iKintosh) [reply etna/analysis/plotters.py:1342]: Nothing will happen, but I will add it as a test case.

File: etna/analysis/plotters.py [hunk 6]
```diff
@@ -1312,12 +1322,47 @@ def _create_holidays_df(country_holidays: Type["holidays_lib.HolidayBase"], time
...
+            ds_upper_bound = pd.timedelta_range(start=0, periods=periods, freq=freq)
+            for bound in ds_upper_bound:
+                ds_add = ds + bound
+                dt.append(ds_add)
+        if "lower_window" in holidays.columns:
```
[the remark this answers is not in the document]
Comment (iKintosh) [reply etna/analysis/plotters.py:1347]: Sorry, my mistake

File: tests/test_analysis/test_plotters.py [hunk 7]
```diff
@@ -197,3 +199,88 @@ def test_validate_intersecting_segments_ok(fold_numbers):
...
+    classic_df = generate_ar_df(periods=30, start_time="2020-01-01", n_segments=1, freq="H")
+    ts = TSDataset.to_dataset(classic_df)
+    holidays = pd.DataFrame({"holiday": "Christmas", "ds": pd.to_datetime(["2020-01-01"]), "upper_window": 3})
```
[the remark this answers is not in the document]
Comment (iKintosh) [reply tests/test_analysis/test_plotters.py:268]: That a good idea, will do

File: tests/test_analysis/test_plotters.py [hunk 8]
```diff
@@ -197,3 +199,88 @@ def test_validate_intersecting_segments_ok(fold_numbers):
...
+
+
+def test_create_holidays_df_upper_window(simple_df):
+    holidays = pd.DataFrame({"holiday": "New Year", "ds": pd.to_datetime(["2020-01-01"]), "upper_window": 2})
+    df = _create_holidays_df(holidays, simple_df.index, as_is=False)
+    assert df.sum().sum() == 3
+
+
+def test_create_holidays_df_upper_window_only(simple_df):
+    """Test if upper_window bounds are used even in case where holiday and TSDataset do not intersect."""
```
[the remark this answers is not in the document]
Comment (alex-hse-repository) [reply tests/test_analysis/test_plotters.py:245]: Lets name it "test_create_holidays_df_upper_window_out_of_index" and remove doctoring

File: etna/analysis/plotters.py [hunk 9]
```diff
@@ -1315,12 +1325,51 @@ def _create_holidays_df(country_holidays: Type["holidays_lib.HolidayBase"], time
...
+            if periods < 0:
+                raise ValueError("Upper windows should be positive.")
+            ds_upper_bound = pd.timedelta_range(start=0, periods=periods + 1, freq=freq)
+            for bound in ds_upper_bound:
+                ds_add = ds + bound
+                dt.append(ds_add)
+        if "lower_window" in holidays.columns:
+            periods = holidays[holidays["holiday"] == name]["lower_window"].fillna(0).tolist()[0]
+            if periods > 0:
+                raise ValueError("Lower windows should be negative.")
```
Comment (alex-hse-repository) [inline etna/analysis/plotters.py:1355]: Non-positive

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

**Review by alex-hse-repository — DISMISSED** [review]
👍

**Pull request merged**